modules/core/src/smithy4s/schema/DefaultValueSchemaVisitor.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+package schema
+
+import smithy4s.schema.Primitive._
+
+trait GetDefaultValue[A] { self =>
+  def getDefault: Option[Document]
+}
+
+object GetDefaultValue {
+  def none[A]: GetDefaultValue[A] = new GetDefaultValue[A] {
+    def getDefault: Option[Document] = None
+  }
+}
+
+object DefaultValueSchemaVisitor extends SchemaVisitor[GetDefaultValue] {
+
+  private def instance[A](maybeDefault: Option[Document]): GetDefaultValue[A] =
+    new GetDefaultValue[A] {
+      def getDefault: Option[Document] = maybeDefault
+    }
+
+  private def from[A](
+      hints: Hints,
+      onNullDefault: => Option[Document]
+  ): GetDefaultValue[A] = {
+    hints.get(smithy.api.Default).map(_.value) match {
+      case Some(d) if d == Document.DNull => instance(onNullDefault)
+      case d                              => instance(d)
+    }
+  }
+
+  def primitive[P](
+      shapeId: ShapeId,
+      hints: Hints,
+      tag: Primitive[P]
+  ): GetDefaultValue[P] = from(
+    hints,
+    tag match {
+      case PShort      => Some(Document.fromInt(0))
+      case PString     => Some(Document.fromString(""))
+      case PFloat      => Some(Document.fromDouble(0))
+      case PDouble     => Some(Document.fromDouble(0))
+      case PTimestamp  => None
```

Comment:
The Timestamp also has a default ```timestamp: 0 seconds since the Unix epoch```
https://github.com/awslabs/smithy/blob/ee0b4ff90daaaa5101f32da936c25af8c91cc6e9/designs/defaults-and-model-evolution.md